### DISCUSSION TITLE
When will Directus be available in PHP / Laravel?

### DISCUSSION THREAD

**Discussion by muhrusdi** [directus/directus#4407] [Archive]
When directus 9 available in laravel?

**Comment by ghost** [comment]
@muhrusdi Directus 9 is a NodeJS project, not PHP. So it can't be installed in laravel. What you can do is use Laravel to interact with Directus 9 using the Rest/GraphQL API. https://docs.directus.io/reference/api/introduction/

**Reply by muhrusdi** [reply]
![Screenshot_20210307-055905_Chrome](https://user-images.githubusercontent.com/9457288/110222002-48a36380-7f0a-11eb-838f-4fa308240f9d.jpg)
How about this?

**Reply by benhaynes** [reply]
Yup! There is a PHP version of Directus 9 in development. It is about 80% complete, but we've paused work on this until the node version is formally released. The node version will be the authoritative spec for Directus. No ETA...

**Reply by ghost** [reply]
My bad. That's good news, I didn't know a Laravel version is in development. The goal is to have both (node + laravel) cores available, and dev chooses which one to connect to vuejs? How did you imagine it?

**Reply by benhaynes** [reply]
Yup! They would both be bundled with the same vue.js app. Since the API is relatively static, we would hope to get featured parity internally, and then have the node version b officially maintained, and the PHP version more of a community project.

**Reply by ghost** [reply]
@benhaynes Is there any repo available to try out or see progress ? Thank you.

**Reply by muhrusdi** [reply]
I use directus v8 in my stack, then i deploy on shared hosting

**Reply by benhaynes** [reply]
@zenati — as of now the Laravel repo is private so as not to cause confusion or give the impression that it is ready. Progress has been paused until v9 Node is out in production.

**Reply by danielbidala** [reply]
I'm really looking forward to the Laravel version. Our primary framework for web apps (CRM, ERP) is laravel. Laravel has a huge community and I think the laravel version will bring a lot of developers to Directus. There are plenty of admin apps for Laravel but none of them compete with Directus. Maybe except Laravel Nova, but it's not an open source project.

**Reply by besucher80** [reply]
@benhaynes you said that the Repo is private at the Moment . Is there a possibilty to Support the Development ? Or have a Look at the Repo to See currenr Status ? This would be really helpful. Thanks for the great work of you all

**Reply by benhaynes** [reply]
The repo is private and in an indeterminate state. It's up to @rijkvanzanten if we would make that public on the community org. Full transparency, our core team will not be working on or maintaining any other ports of the codebase.

**Reply by rijkvanzanten** [reply]
That repo is not in a state where it could be open-sourced. It's architected very differently from our final approach, and is no where near feature complete with the v9 prod release. Releasing it in any (semi-) official capacity now would do more harm than good.

**Reply by ricardovigatti** [reply]
Man, Directus works so well in NodeJS, makes no sense to keep parallel development of it in PHP. NodeJS is easy to learn, easy to develop custom code, easy to host.

lol

I wouldn't worry with PHP at all.

**Reply by rijkvanzanten** [reply]
@ricardovigatti When Docker wasn't really a thing yet, there were some serious ideas to support both "stacks" (eg traditional LAMP and whatever you want to call modern node these days), but now that everything Directus is cloud native, there (like you correctly stated) really isn't a point to bother with multiple supported runtimes, as the exact runtime started to matter less.

**Reply by vitalijalbu** [reply]
hi all, can you share the existing laravel project with me? or maybe i can find it somewhere?? thanks
